src/backend/columnar/columnar_customscan.c

```diff
@@ -660,6 +668,9 @@ ColumnarScan_CreateCustomScanState(CustomScan *cscan)
 		columnarScanState->qual = cscan->scan.plan.qual;
 	}
 
+	/* used by ColumnarVarNeeded for EXPLAIN when building Var's from scratch */
+	columnarScanState->scanrelid = cscan->scan.scanrelid;
+
```

Comment:
It looks like you found a place we already have this value -- we actually don't need to store it again. In ColumnarScan_ExplainCustomScan(), all you need to do is something like `((CustomScan *)node->ss.ps.plan)->scan->scanrelid`.